Handle http->https redirect internal ip disclosure. Also refactored a…

coryleeio opened the pull request:
…nd cleaned up nginx config.

Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).

coryleeio commented:
Mostly moved each shared concern into of the HTTP and HTTPS server into a concerns/ template directory.  I have validated that the templates are valid NGINX templates on stage for ecom/discovery/credentials/veda_frontend in docker for both NGINX_ENABLE_SSL=true and NGINX_ENABLE_SSL=false

The only real changes in behaviour are the following:
     * no default server is set bc in NGINX the first server is automatically the default one and this lets me delete some code, effectively no change.
     * added an error when you are going to accidentally disclose your internal IP during HTTP->HTTPS redirect.
     * Always add cross-origin header so that error pages still get the cross-origin header.

feanil commented:
Do you have a sandbox where this is deployed?

feanil commented on playbooks/roles/nginx/templates/edx/app/nginx/sites-available/insights.j2:
```diff
@@ -37,7 +37,6 @@ server {
   location ~ ^/static/(?P<file>.*) {
     root {{ COMMON_DATA_DIR }}/{{ insights_service_name }};
     add_header Cache-Control "max-age=31536000";
-    add_header 'Access-Control-Allow-Origin' $cors_origin;
```
This is the only change made in the insights template, is this on purpose?

coryleeio replied:
No, good catch.  Thanks.  I didn't mean to affect the nginx role. I'll fix this

feanil commented on playbooks/roles/edx_django_service/templates/edx/app/nginx/sites-available/concerns/cross-origin.j2:
```diff
@@ -0,0 +1,17 @@
+  "~*^https?:\/\/{{ host|replace('.', '\.') }}$" $http_origin;
+{% endfor %}
+}
+add_header 'Access-Control-Allow-Origin' $cors_origin always;
```
Shouldn't hurt but note that this will now be on all responses including the HTML responses, not just the static assets.

coryleeio commented:
@feanil I have removed the cors always header as we discussed.  I have also removed the accidental change to the nginx role.  It is stood up on a sandbox at my url:
https://coryleeio.sandbox.edx.org

It does appear to be working.

coryleeio commented:
@feanil Sandbox is back up
https://coryleeio.sandbox.edx.org

coryleeio commented:
contingent upon pass of: https://tools-edx-jenkins.edx.org/job/Sandboxes/job/CreateSandbox/10460/
